updater/bin/update-script.rb

```diff
@@ -554,8 +554,8 @@ def show_diff(original_file, updated_file)
   credentials: $options[:credentials]
 )
 user_id = azure_client.get_user_id
-default_branch_name = azure_client.fetch_default_branch($source.repo)
-active_pull_requests = azure_client.pull_requests_active(user_id, default_branch_name)
+$options[:branch] = azure_client.fetch_default_branch($source.repo) unless $options[:branch]
```

Comment:
```diff
- $options[:branch] = azure_client.fetch_default_branch($source.repo) unless $options[:branch]
+ target_branch = $options[:branch] || azure_client.fetch_default_branch($source.repo)
```

Reply:
Creating a local variable and so creating a new reference in memory seems like a waste of resources, since you already have a pointer that was intended for this purpose.

After execution the `branch` property is not going to be changed by the end user anymore and there is no further mutation being done on `target_branch`, so I reckon this unnecessarily costs memory and CPU cycles.

Reply:
@mburumaxwell any feedback on the above?

Reply:
The preference is to not do any changes to the `$options` object after the initial setup.